### PR TITLE
Add support for GPTQ using native transformers/peft

### PR DESCRIPTION
**Pull request by winglian**

### CODE REVIEW

File: examples/gptq/lora.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+base_model: TheBloke/Llama-2-7B-GPTQ
```
Comment (NanoCode012) [inline examples/gptq/lora.yml]: this file should maybe be in `examples/llama2/gptq-lora.yml`

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,5 @@
+--extra-index-url https://huggingface.github.io/autogptq-index/whl/cu117/
```
Comment (NanoCode012) [inline requirements.txt]: is this cuda version ok?

File: src/axolotl/utils/models.py [hunk 3]
```diff
@@ -147,32 +148,15 @@ def load_model(
...
-            # For backward compatibility
-            from peft import (
-                prepare_model_for_int8_training as prepare_model_for_kbit_training,
-            )
-
     model_kwargs = {}
     if cfg.model_revision:
         model_kwargs["revision"] = cfg.model_revision
+    if cfg.gptq:
+        # TODO we should figure out how read the models config.json first
```
Comment (NanoCode012) [inline src/axolotl/utils/models.py]: We can load from config with AutoConfig.from_pretrained?

Comment (winglian) [reply src/axolotl/utils/models.py]: yeah, I might tackle this in a refactor. not sure how easily this is handled for cases where it points to a local model path.

**Pull request merged**